lib/src/main/java/org/esbtools/eventhandler/Result.java

```diff
@@ -18,10 +18,7 @@
 
 package org.esbtools.eventhandler;
 
-public interface LookupResult {
-    Object getBody();
-
-    boolean hasErrors();
-
-    DocumentEvent getAssociatedEvent();
+public interface Result<T> {
+    T get();
```

Comment:
We could rename if you would like. I avoided the JDK Future interface since it is much more complicated to implement. The new `CompletableFuture` from JDK8 gets close but I would still have to override `get()` to do the lazy fetching of responses so it gets a little weird.